### ISSUE TITLE
Login is not blocked after multiple retries

### ISSUE THREAD

**Issue by saw-jan** [owncloud/brute_force_protection#191]
### Steps to reproduce
1. Pull latest core master (https://github.com/owncloud/core)
2. Checkout to commit [eaf5a9fe952b2780a053d2b17026e420c6542381](https://github.com/owncloud/core/commit/eaf5a9fe952b2780a053d2b17026e420c6542381)
3. Run: `make clean` and `make`
4. Enable `brute_force_protection` app
```bash
sudo -u www-data ./occ a:e brute_force_protection
```
5. Set the these from `Settings` -> `[Admin] Security`
![Screenshot from 2023-01-06 14-47-17](https://user-images.githubusercontent.com/52366632/210967850-b18b6b3e-1518-4c2d-a81d-cdffefac050f.png)

6. Try login with invalid password many times

### Expected
to see login blocked screen
![Screenshot from 2023-01-06 14-49-52](https://user-images.githubusercontent.com/52366632/210968139-e574611d-a138-46e1-b983-d0d575e66610.png)

### Actual
No login blocked screen
Users can try to login infinitely

#### Details
The behavior occurred after PR https://github.com/owncloud/core/pull/40520

**Comment by phil-davis** [comment]
@jvillafanez @DeepDiver1975 I guess that this is a regression related to https://github.com/owncloud/core/pull/40520

**Comment by jvillafanez** [comment]
Yes, this is a change in the behavior caused by that PR.

Right now, with the changes in core, we first check whether the password is correct or not. If the password is wrong, we don't check the brute force protection and fail the login right away. Only when the password is right we check for the brute force protection.

Technically, the login is blocked because the login attempt might still fail due to the brute force protection (assuming it's still in the cool-down period) even if the password is right. However, we might still need to change the behavior because we leak the password.

Summing up, right now, the message appears only if the password is the correct one, otherwise, the login will fail without any particular message

**Comment by saw-jan** [comment]
I tried manually today, now it's working just as expected. Didn't know what happened previously.
I will have a look at the test code.

**Comment by saw-jan** [comment]
> Summing up, right now, the message appears only if the password is the correct one, otherwise, the login will fail without any particular message

So now, the blocking behavior has changed. Previously, trying to attempt login with an invalid password after a certain threshold also blocked the login. But now, after reaching the threshold, trying with valid login will block the login.

If that's the correct intended behavior then we need to fix the test code.

CC @jvillafanez @phil-davis

**Comment by jvillafanez** [comment]
> If that's the correct intended behavior then we need to fix the test code.

I'm not sure if this is what we want. An attacker could figure out the password, so even if he can't access to the account right now, he could after the banning period is over.
This didn't happen before because the attacker couldn't figure out the password

**Comment by jvillafanez** [comment]
Can someone check if https://github.com/owncloud/brute_force_protection/pull/193 really fix this issue?
As said in the PR, the behavior should be the same as before, and the code should be backwards-compatible.

**Comment by saw-jan** [comment]
as per the pipeline: https://drone.owncloud.com/owncloud/brute_force_protection/2204/12/13
the expected test scenario failed but was not in the expected step.
I will have a look on my machine.
